src/UglyToad.PdfPig/Export/AltoXmlTextExporter.cs

```diff
@@ -0,0 +1,5337 @@
+            };
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        private AltoDocument.AltoDescription GetAltoDescription(string fileName)
+        {
+            var processing = new AltoDocument.AltoDescriptionProcessing()
+            {
+                ProcessingAgency = null,
+                ProcessingCategory = AltoDocument.AltoProcessingCategory.Other, // TBD
+                ProcessingDateTime = DateTime.UtcNow.ToString(),
+                ProcessingSoftware = new AltoDocument.AltoProcessingSoftware()
+                {
+                    SoftwareName = "PdfPig",
+                    SoftwareCreator = @"https://github.com/UglyToad/PdfPig",
+                    ApplicationDescription = "Read and extract text and other content from PDFs in C# (port of PdfBox)",
+                    SoftwareVersion = "x.x.xx"
+                },
+                ProcessingStepDescription = null,
+                ProcessingStepSettings = pageSegmenter.GetType().Name + "|" + wordExtractor.GetType().Name,
+                Id = "P" + pageCount + "_D1"
+            };
+
+            var documentIdentifier = new AltoDocument.AltoDocumentIdentifier()
+            {
+                DocumentIdentifierLocation = null,
+                Value = null
+            };
+
+            var fileIdentifier = new AltoDocument.AltoFileIdentifier()
+            {
+                FileIdentifierLocation = null,
+                Value = null
+            };
+
+            return new AltoDocument.AltoDescription()
+            {
+                MeasurementUnit = AltoDocument.AltoMeasurementUnit.Pixel, // need to check that
+                Processings = new[] { processing },
+                SourceImageInformation = new AltoDocument.AltoSourceImageInformation()
+                {
+                    DocumentIdentifiers = new AltoDocument.AltoDocumentIdentifier[] { documentIdentifier },
+                    FileIdentifiers = new AltoDocument.AltoFileIdentifier[] { fileIdentifier },
+                    FileName = fileName
+                }
+            };
+        }
+
+        private string Serialize(AltoDocument altoDocument)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(AltoDocument));
+            var settings = new XmlWriterSettings()
+            {
+                //Encoding = new System.Text.UTF8Encoding(true),
+                Indent = true,
+                IndentChars = indentChar,
+                OmitXmlDeclaration = true // hack to manually handle utf-8
+            };
+
+            using (var stringWriter = new System.IO.StringWriter())
+            using (var xmlWriter = XmlWriter.Create(stringWriter, settings))
+            {
+                stringWriter.WriteLine("<?xml version=\"1.0\" encoding=\"UTF-8\"?>"); // hack to manually handle utf-8
```

Comment:
Could something like this work better for this purpose? https://timvw.be/2007/01/08/generating-utf-8-with-systemxmlxmlwriter/

Does the Alto format require UTF8?

Reply:
It seems they all need UTF8. Will look at your link